### PR TITLE
Coinbase component test

### PR DESCRIPTION
**Pull request by KaustubhShamshery**
## Description
Coinbase test

1. Checks the lookup percentage
2. Checks the percentage normal nodes are rewarded
3. checks  the total percentage with the total reward

### Implementation

- [x] **ready for review**

### Integration Test (Core Team)

- [x] local machine test

- [ ] small-scale cloud test

### CODE REVIEW

File: src/libLookup/Lookup.h [hunk 1]
```diff
@@ -153,6 +153,8 @@ class Lookup : public Executable {
   // Hardcoded for now -- to be called by constructor
   void SetLookupNodes();
 
+  void SetLookupNodes(VectorOfNode);
```
Comment (ansnunez) [inline src/libLookup/Lookup.h]: Can change to `VectorOfNode&`

File: tests/Incentives/test_coinbase.cpp [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+      b1 = GenerateRandomBooleanVector(shard.size());
+
+      b2 = GenerateRandomBooleanVector(shard.size());
+      dummyDS.SaveCoinbaseCore(b1, b2, shard, j++, i + 1);
+    }
+
+    b1 = GenerateRandomBooleanVector(dummy_ds_comm.size());
+    b2 = GenerateRandomBooleanVector(dummy_ds_comm.size());
+
+    dummyDS.SaveCoinbaseCore(b1, b2, dummy_ds_comm,
```
Comment (ansnunez) [inline tests/Incentives/test_coinbase.cpp]: Do you also need to add `CoinbaseReward::LOOKUP_REWARD` shard?

Comment (KaustubhShamshery) [reply tests/Incentives/test_coinbase.cpp]: InitCoinbase rewards them equally, so no  need

File: src/libLookup/Lookup.h (hunk 1 above)
Comment (KaustubhShamshery) [reply src/libLookup/Lookup.h]: OK

Comment (KaustubhShamshery) [reply src/libLookup/Lookup.h]: OK

**Pull request merged**